### PR TITLE
OAuth2 configuration redesign

### PR DESCRIPTION
**Pull request by dashevchenko**
## Pull Request description

The existing oauth2 configuration was broken into 3 controllers: DomainController, MobileAppController, Oauth2Controller.

## General checklist

- [x] You have reviewed the guidelines [document](https://docs.google.com/document/d/1wqcOafLx5hth8SAg4dqV_LV3un3m5WYR8RdTJ4MbbUM/edit?usp=sharing).
- [x] [Labels](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels#about-labels) that classify your pull request have been added.
- [x] The [milestone](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/about-milestones) is specified and corresponds to fix version.  
- [x] Description references specific [issue](https://github.com/thingsboard/thingsboard/issues).
- [x] Description contains human-readable scope of changes.
- [x] Description contains brief notes about what needs to be added to the documentation.
- [x] No merge conflicts, commented blocks of code, code formatting issues.
- [x] Changes are backward compatible or upgrade script is provided.
- [x] Similar PR is opened for PE version to simplify merge. Crosslinks between PRs added. Required for internal contributors only.
  
## Front-End feature checklist

- [ ] Screenshots with affected component(s) are added. The best option is to provide 2 screens: before and after changes;
- [ ] If you change the widget or other API, ensure it is backward-compatible or upgrade script is present.
- [ ] Ensure new API is documented [here](https://github.com/thingsboard/thingsboard-ui-help)

## Back-End feature checklist

- [x] Added corresponding unit and/or integration test(s). Provide written explanation in the PR description if you have failed to add tests.
- [ ] If new dependency was added: the dependency tree is checked for conflicts.
- [ ] If new service was added: the service is marked with corresponding @TbCoreComponent, @TbRuleEngineComponent, @TbTransportComponent, etc.
- [x] If new REST API was added: the RestClient.java was updated, issue for [Python REST client](https://github.com/thingsboard/thingsboard-python-rest-client) is created.
- [ ] If new yml property was added: make sure a description is added (above or near the property).

### CODE REVIEW

File: common/data/src/main/java/org/thingsboard/server/common/data/audit/ActionType.java [hunk 1]
```diff
@@ -57,7 +57,8 @@ public enum ActionType {
     ADDED_COMMENT(false, TbMsgType.COMMENT_CREATED),
     UPDATED_COMMENT(false, TbMsgType.COMMENT_UPDATED),
     DELETED_COMMENT(false, null),
-    SMS_SENT(false, null);
+    SMS_SENT(false, null),
+    UPDATED_OAUTH2_CLIENTS(false, null);
```
Comment (ViacheslavKlimov) [inline common/data/src/main/java/org/thingsboard/server/common/data/audit/ActionType.java:61]: I think this is redundant, we might not log this at all, only the domain update

File: ui-ngx/src/app/modules/home/pages/admin/oauth2/clients/client-list.component.ts
Comment (vvlladd28) [inline ui-ngx/src/app/modules/home/pages/admin/oauth2/clients/client-list.component.ts:1]: Maybe need to use tb-entity-list

**Review by vvlladd28 — CHANGES_REQUESTED** [review]

**Review by vvlladd28 — APPROVED** [review]

**Review by ViacheslavKlimov — CHANGES_REQUESTED** [review]

File: application/src/main/data/upgrade/3.7.0/schema_update.sql [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+                (SELECT d.id, r.id FROM domain d LEFT JOIN oauth2_client r on d.oauth2_params_id = r.oauth2_params_id
+                                   WHERE r.platforms IS NULL OR r.platforms IN ('','WEB'));
+
+            INSERT INTO mobile_app_oauth2_client(mobile_app_id, oauth2_client_id)
+                (SELECT m.id, r.id FROM mobile_app m LEFT JOIN oauth2_client r on m.oauth2_params_id = r.oauth2_params_id
+                                   WHERE r.platforms IS NULL OR r.platforms IN ('','ANDROID','IOS'));
+
+            ALTER TABLE mobile_app RENAME CONSTRAINT oauth2_mobile_pkey TO mobile_app_pkey;
+            ALTER TABLE domain RENAME CONSTRAINT oauth2_domain_pkey TO domain_pkey;
+            ALTER TABLE oauth2_client RENAME CONSTRAINT oauth2_registration_pkey TO oauth2_client_pkey;
```
[the remark this answers is not in the document]
Comment (dashevchenko) [reply application/src/main/data/upgrade/3.7.0/schema_update.sql:77]: yes, all other will be dropped after column removal

File: application/src/main/java/org/thingsboard/server/controller/OAuth2Controller.java [hunk 3]
```diff
@@ -85,37 +100,80 @@ public List<OAuth2ClientInfo> getOAuth2Clients(HttpServletRequest request,
...
-        return oAuth2Service.findOAuth2Info();
+    @GetMapping(value = "/oauth2/client/infos", params = {"clientIds"})
+    public List<OAuth2ClientInfo> findTenantOAuth2ClientInfosByIds(
+            @Parameter(description = "A list of oauth2 ids, separated by comma ','", array = @ArraySchema(schema = @Schema(type = "string")), required = true)
+            @RequestParam("clientIds") String[] strClientIds) throws ThingsboardException {
+        checkArrayParameter("clientIds", strClientIds);
+        List<OAuth2ClientId> oAuth2ClientIds = new ArrayList<>();
+        for (String oauth2ClientId : strClientIds) {
+            oAuth2ClientIds.add(new OAuth2ClientId(toUUID(oauth2ClientId)));
+        }
```
[the remark this answers is not in the document]
Comment (dashevchenko) [reply application/src/main/java/org/thingsboard/server/controller/OAuth2Controller.java:148]: getOAuth2ClientIds check read permission for each client, so I will just refactor this code

File: application/src/main/java/org/thingsboard/server/service/edge/EdgeContextComponent.java [hunk 4]
```diff
@@ -167,7 +167,7 @@ public class EdgeContextComponent {
     private NotificationTemplateService notificationTemplateService;
 
     @Autowired
-    private OAuth2Service oAuth2Service;
+    private OAuth2ClientService oAuth2ClientService;
```
[the remark this answers is not in the document]
Comment (dashevchenko) [reply application/src/main/java/org/thingsboard/server/service/edge/EdgeContextComponent.java:170]: @AndriiLandiak could you please take a look

File: dao/src/main/java/org/thingsboard/server/dao/oauth2/OAuth2ClientDao.java [hunk 5]
```diff
@@ -15,20 +15,32 @@
...
-public interface OAuth2RegistrationDao extends Dao<OAuth2Registration> {
+public interface OAuth2ClientDao extends Dao<OAuth2Client> {
 
-    List<OAuth2Registration> findEnabledByDomainSchemesDomainNameAndPkgNameAndPlatformType(List<SchemeType> domainSchemes, String domainName, String pkgName, PlatformType platformType);
+    PageData<OAuth2Client> findByTenantId(UUID tenantId, PageLink pageLink);
 
-    List<OAuth2Registration> findByOAuth2ParamsId(UUID oauth2ParamsId);
+    List<OAuth2Client> findEnabledByDomainName(String domainName);
+
+    List<OAuth2Client> findEnabledByPckNameAndPlatformType(String pkgName, PlatformType platformType);
```
[the remark this answers is not in the document]
Comment (dashevchenko) [reply dao/src/main/java/org/thingsboard/server/dao/oauth2/OAuth2ClientDao.java:35]: I forgot enabled in query, will fix

**Review by ViacheslavKlimov — APPROVED** [review]

**Review by vvlladd28 — APPROVED** [review]

**Pull request merged**